### DISCUSSION TITLE
colornames.ini

### DISCUSSION THREAD

**Discussion by raonath** [xournalpp/xournalpp#5363] [Q&A]
I am using Xournalpp 1.1.3 buit on 11/23/2022 on MX Linux (a Debian based distro). In my configuration folder, there is a file named colornames.ini that has only two lines:
[info]
about=Xournalpp custom color names
I thought this was to name custom colors I put in toolbar.ini. I am unable to find how to set this up though.

Thanks in advance for your help.
Nath Rao

**Comment by rolandlo** [comment]
In version 1.1.3 the file `colornames.ini` was used to translate from color rgb values to strings in your language. I don't quite remember the syntax, but one could take a look [here](https://github.com/xournalpp/xournalpp/pull/2379/files#diff-0d5ed1bcf0bb37c261e3e6439809a1b040c391ab0ee5fe764cab02b640de738e) to figure it out.

In version 1.2.0 custom color palette support has been added. You can define your colors in a `palette.gpl` file and put it into the config folder (`~/.config/xournalpp/` in case of Linux). 

Example:
```.gpl
GIMP Palette
Name: Xournal New Palette.gpl
#
255 225 107 gelb
255 161  84 orange
205 158 247 lila
155 219  77 hellgrün
100 186 255 hellblau
128 128 128 grau
 58 145   4 dunkelgrün
237  83  83 rot
  0  46 153 dunkelblau
  0   0   0 schwarz
255 255 255 weiss
```
In toolbar.ini those colors are then simply referenced by `COLOR(0)`, `COLOR(1)` etc.
So consider upgrading to the latest version and creating a `palette.gpl` that fits your needs.

**Comment by raonath** [comment]
Thanks. I thought that colornames.ini was to translate FGB values in
toolbar.ini to human readable names but could not get it to work. I will
look at the link you give.
The online documentation talks only about palette.gpl, but says that it is
available only in the nightly builds.  Anyway, I would rather stay with
what is in the distro, and this irritation is not enough to compile and
install myself repeatedly. [This is for an anemic netbook that I may have
to use for presentations/lectures etc.]

Thanks
Nath Rao

**Reply by rolandlo** [reply]
> The online documentation talks only about palette.gpl, but says that it is available only in the nightly builds.

The website was outdated about that. I have removed that note.

You don't need to compile and install yourself even if your distro doesn't have up to date versions and you're not on Ubuntu. Instead you could use one of the packaged formats (like the flatpak, snap package or AppImage).
